test/integration/node-specific/mongo_client.test.ts

```diff
@@ -649,5 +650,26 @@ describe('class MongoClient', function () {
       expect(endEvents).to.have.lengthOf(1);
       expect(endEvents[0]).to.have.property('reply', undefined); // noReponse: true
     });
+
+    context('when server selection would return no servers', () => {
```

Comment:
Given the other comment about possibly undefined topology, it would be good to also cover that edge case, too, if we can

Reply:
The test case to cover this would be quite involved to get the driver in an internal state where the topology was actually null at that point in the code. The only situation where it could happen would be simultaneous async calls to `close()` where one had gotten to the point of closing the topology before the other one got to the point of ending sessions. I started to try and write a test and it just got way out of hand trying to mimic that behaviour in it.

Reply:
ok, given it's an edge case I think it's not a big deal, and we're just following the pattern in the .then chain

leaving the comment open for visibility